### PR TITLE
WIP Netsuite: skip fetching typesToSkip also in SDF according to fetchAllTypesAtOnce, support config changes suggestions

### PR DESCRIPTION
**Pull request by omrilit**

### CODE REVIEW

File: packages/netsuite-adapter/src/adapter.ts [hunk 1]
```diff
@@ -88,15 +94,20 @@ export default class NetsuiteAdapter implements AdapterOperations {
    * Account credentials were given in the constructor.
    */
   public async fetch(): Promise<FetchResult> {
-    const customObjects = this.client.listCustomObjects().catch(e => {
-      log.error('failed to list custom objects. reason: %o', e)
-      return [] as CustomizationInfo[]
-    })
+    const customTypesToFetch = _.pull(Object.keys(customTypes), ...this.typesToSkip)
+    const listCustomObjectsResult = this.client
```
Comment (adamro) [inline packages/netsuite-adapter/src/adapter.ts:98]: what's the use of `listCustomObjectsResult`?

why not just do 

```
const {
      elements: customObjects,
      failedTypes,
      failedToFetchAllAtOnce,
    } = await this.client.listCustomObjects(customTypesToFetch, this.fetchAllTypesAtOnce)
```
at line 105?

File: packages/netsuite-adapter/src/client/client.ts [hunk 2]
```diff
@@ -315,29 +322,80 @@ export default class NetsuiteClient {
...
-    const project = await this.initProject()
-    await NetsuiteClient.executeProjectAction(COMMANDS.IMPORT_OBJECTS, {
-      destinationfolder: `${SDF_PATH_SEPARATOR}${OBJECTS_DIR}`,
-      type: 'ALL',
-      scriptid: 'ALL',
-      excludefiles: true,
-    }, project.executor)
-
-    const objectsDirPath = NetsuiteClient.getObjectsDirPath(project.projectName)
+  async listCustomObjects(typeNames: string[], fetchAllAtOnce: boolean):
```
Comment (adamro) [inline packages/netsuite-adapter/src/client/client.ts:325]: nit and unrelated to this PR but why is this called `listCustomObjects` and not `getCustomObjects`? 
`listCustomObjects` sound like it's listing their names and not actually getting them

File: packages/netsuite-adapter/src/client/client.ts [hunk 3]
```diff
@@ -315,29 +322,80 @@ export default class NetsuiteClient {
...
+    Promise<ListCustomObjectsResult> {
+    const { executor, projectName } = await this.initProject()
+    const { failedToFetchAllAtOnce, failedTypes } = await NetsuiteClient.importObjects(typeNames,
```
Comment (adamro) [inline packages/netsuite-adapter/src/client/client.ts:328]: `failedToFetchAllAtOnce` is redundant.
`failedToFetchAllAtOnce === (failedTypes.length > 0 && fetchAllAtOnce)`

So you can change `NetsuiteClient.importObjects`'s return type to be only failedTypes (string[]).

File: packages/netsuite-adapter/src/client/client.ts [hunk 4]
```diff
@@ -315,29 +322,80 @@ export default class NetsuiteClient {
...
+        }
+        const additionalFileContent = readFile(osPath.resolve(objectsDirPath, additionalFilename))
+        return convertToTemplateCustomTypeInfo((await xmlContent).toString(), scriptId,
+          additionalFilename.split(FILE_SEPARATOR)[2], (await additionalFileContent).toString())
+      })
+    )
+    return { elements, failedTypes, failedToFetchAllAtOnce }
+  }
+
+  private static async importObjects(typeNames: string[], fetchAllAtOnce: boolean,
```
Comment (adamro) [inline packages/netsuite-adapter/src/client/client.ts:349]: If you go with my comment about and remove the `failedToFetchAllAtOnce`, I would change the func a bit.

```
private static async importObjects(typeName: string[], fetchAllAtOnce: boolean, executor: CommandActionExecutorType): Promise<string[]> {
const getAllAtOnce = async (): boolean => {
     log.debug('Fetching all custom objects at once')
      try {
        await NetsuiteClient.runImportObjectsCommand('ALL', executor)
        return true
      } catch (e) {
        log.warn(`Attempt to fetch all custom objects has failed due to: ${e}`)
        return false
}

if ( fetchAllAtOnce && await getAllAtOnce()) {
    return []
}
log.warn(`Fetching objects type by type`)
return await NetsuiteClient.importObjectsByTypes(typeNames, executor)
}```

File: packages/netsuite-adapter/src/client/client.ts [hunk 5]
```diff
@@ -315,29 +322,80 @@ export default class NetsuiteClient {
...
+    )
+    return failedTypes
+  }
+
+  private static async runImportObjectsCommand(type: string, executor: CommandActionExecutorType):
+    Promise<OperationResult> {
+    return NetsuiteClient.executeProjectAction(COMMANDS.IMPORT_OBJECTS, {
+      destinationfolder: `${SDF_PATH_SEPARATOR}${OBJECTS_DIR}`,
+      type,
+      scriptid: 'ALL',
```
Comment (adamro) [inline packages/netsuite-adapter/src/client/client.ts:396]: nit: const 'ALL' maybe? you use it here and at line 355

File: packages/netsuite-adapter/src/config.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  },
+})
+
+export type NetsuiteConfig = {
+  [TYPES_TO_SKIP]?: string[]
+  [FILE_PATHS_REGEX_SKIP_LIST]?: string[]
+  [FETCH_ALL_TYPES_AT_ONCE]?: boolean
+}
+
+export const STOP_MANAGING_ITEMS_MSG = 'Salto failed to fetch some items from netsuite. '
```
Comment (adamro) [inline packages/netsuite-adapter/src/config.ts:49]: nit: Usually appropriate to use "cased" name when referring to a company -  `NetSuite`

File: packages/netsuite-adapter/src/config.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  + 'In order to complete the fetch operation, '
+  + 'Salto needs to stop managing these items by applying the following configuration change:'
```
Comment (adamro) [inline packages/netsuite-adapter/src/config.ts:51]: I know you used the Salesforce copy but I would ask for an additional review probably from @tomermevorach 

My take - Too much `Salto` in the copy for my liking :)

File: packages/netsuite-adapter/src/adapter.ts (hunk 1 above)
Comment (omrilit) [reply packages/netsuite-adapter/src/adapter.ts:98]: The reason is to parallelize the calls to import objects & files

File: packages/netsuite-adapter/src/client/client.ts (hunk 3 above)
Comment (omrilit) [reply packages/netsuite-adapter/src/client/client.ts:328]: As explained f2f, in this case we don't handle the situation of failing with 'ALL' and then passing for all types.
decided to stay with current implementation.

**Pull request merged**